jwst/datamodels/schemas/filteroffset.schema.yaml

```diff
@@ -30,3 +30,5 @@ allOf:
             type: number
           row_offset:
             type: number
+          pupil:
```

Comment:
Now that there are entries for both filter and pupil, would it make sense to rename the existing "filters" attribute to something more generic and change "name" to "filter"?

Reply:
And how does this addition affect the use of this model for MIRI, which only uses filter? Will they have to redeliver their ref file(s) to contain blank/dummy entries for "pupil"? Or is "pupil" somehow an optional attribute?

Reply:
If we don't change `name` to `filter` adding `pupil` is transparent to the MIRI team.

Reply:
I take this back, we can allow both ways if the schema is something like:
```
    filters:
      anyOf:
        - type: array
          items:
            type: object
            properties:
              name:
                type: string
              column_offset:
                type: number
              row_offset:
                type: number
        - type:
          items:
            type: object
            properties:
              filter:
                type: string
              column_offset:
                type: number
              row_offset:
                type: number
              pupil:
                  type: string

```

which will produce files like
```
filters:
- {column_offset: 1, filter_name: F1, pupil: P1, row_offset: 1}
- {column_offset: 2, filter_name: F2, pupil: P2, row_offset: 2}
- {column_offset: 3, filter_name: F3, pupil: P3, row_offset: 3}
```

Another alternative is to write the files as matching lists of values as in 
```
column_offset: [-0.599, -0.143, 0.047, 0.0, -0.261, -1.07, -0.212, -0.287, 0.419,
  -0.076, -0.166, 0.145]
filter_names: [F090W, F115W, F140M, F150W, F158M, F200W, F277W, F356W, F380M, F430M,
  F444W, F480M]
meta: {date: '2020-04-01T08:33:21.293', filename: test1.asdf, model_type: FilteroffsetModel,
  telescope: JWST}
pupil_names: [F480M, F444W, F430M, F380M, F356W, F277W, F200W, F158M, F150W, F140M,
  F115W, F090W]
row_offset: [-0.007, 0.068, 0.016, 0.0, -0.322, 2.198, 0.254, -0.068, -0.327, -0.075,
  -0.003, -0.311]
```
That would be a different schema. IN both cases we can keep MIRI files readable.

I have no preference.

Reply:
I like the idea of the first scheme in which the files for NIRCam and NIRISS have columns of `filter`, `pupil`, `column_offset`, and `row_offset`, while the MIRI file remains the same.